sub-packages/bionemo-esm2/src/bionemo/esm2/run/config_models.py

```diff
@@ -0,0 +1,209 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import importlib
+from pathlib import Path
+from typing import Optional, Type
+
+import torch
+from nemo.utils import logging
+from pydantic import field_serializer, field_validator, model_validator
+
+from bionemo.esm2.data.datamodule import ESMDataModule
+from bionemo.esm2.data.dataset import RandomMaskStrategy
+from bionemo.esm2.data.tokenizer import get_tokenizer
+from bionemo.esm2.model.attention import ESM2DotProductAttention, ESM2TEDotProductAttention
+from bionemo.esm2.model.model import ESM2Config
+from bionemo.llm.model.biobert.model import BiobertSpecOption
+from bionemo.llm.run.config_models import (
+    DataConfig,
+    ExposedModelConfig,
+    MainConfig,
+)
+
+
+class ESM2DataConfig(DataConfig[ESMDataModule]):
+    """ESM2DataConfig is a configuration class for setting up the pre-training data module for ESM2.
+
+    The ESM2DataModule implements the cluster oriented sampling method defined in the ESM2 publication.
+
+    Attributes:
+        train_cluster_path (Path): Path to the training cluster data.
+        train_database_path (Path): Path to the training database.
+        valid_cluster_path (Path): Path to the validation cluster data.
+        valid_database_path (Path): Path to the validation database.
+        micro_batch_size (int): Size of the micro-batch. Default is 8.
+        result_dir (str): Directory to store results. Default is "./results".
+        min_seq_length (int): Minimum sequence length. Default is 128.
+        max_seq_length (int): Maximum sequence length. Default is 128.
+        random_mask_strategy (RandomMaskStrategy): Strategy for random masking. Default is RandomMaskStrategy.ALL_TOKENS.
+        num_dataset_workers (int): Number of workers for the dataset. Default is 0.
+
+    Methods:
+        construct_data_module(global_batch_size: int) -> ESMDataModule:
+            Constructs and returns an ESMDataModule instance with the provided global batch size.
+    """
+
+    train_cluster_path: Path
+    train_database_path: Path
+    valid_cluster_path: Path
+    valid_database_path: Path
+
+    micro_batch_size: int = 8
+    result_dir: str = "./results"
+    min_seq_length: int = 128
+    max_seq_length: int = 128
+    random_mask_strategy: RandomMaskStrategy = RandomMaskStrategy.ALL_TOKENS
+    num_dataset_workers: int = 0
+
+    def construct_data_module(self, global_batch_size: int) -> ESMDataModule:
```

Comment:
its because parallelism settings arent a part of the DataConfig. We handle this in the `train` function, where the `MainConfig` has access to both the micro batch size and the parallelism settings. This is also why the DataConfig has a `construct_data_module` argument. I'm going to update the docstring to answer this question.